File: app/assets/javascripts/models/appointment.js
Assisster.Models.Appointment = Backbone.Model.extend({
  urlRoot: "/api/appointments/",
	
	initialize: function () {
	},

  convertToEvent: function(officeHour) {
    var eventObject = {};
		eventObject.id = this.id;
    eventObject.title = this.escape('title');
    eventObject.start = this.escape('startTime');
    eventObject.end = this.escape('endTime');
		eventObject.email = this.escape('email');
		eventObject.fname = this.escape('fname');
		eventObject.lname = this.escape('lname');
		eventObject.phone_number = this.escape('phone_number');
		eventObject.appointment_status = this.escape('appointment_status');
		
		if (!officeHour && this.get('office_hour')) {
			eventObject.rendering = "background";
		}
		if (eventObject.appointment_status === 'Pending') {
			eventObject.backgroundColor = "#257e4a";
			eventObject.borderColor = "#257e4a";
		}
		
    return eventObject;
  },
	
	date: function () {
		return this.startTime().format("dddd, D MMM");
	},
	
	endTime: function () {
		return moment.utc(this.get('endTime'));
	},
	
	fullName: function () {
		return this.escape('fname') + " " + this.escape("lname");
	},
	
	phoneNumber: function () {
		return this.escape('phone_number').slice(0, 3) + "-" + this.escape('phone_number').slice(3);
	},
	
	startTime: function () {
		return moment.utc(this.get('startTime'));
	},
	
	time: function () {
		return this.startTime().format("HH:mm") + " - " + this.endTime().format("HH:mm");
	},
	
	today: function () {
		var today = moment().format('YYYY-MM-DD');
		return this.startTime().isSame(today, "days") &&
					 (this.get('appointment_status') === "Confirmed");
	},
})

File: app/models/appointment.rb
# == Schema Information
#
# Table name: appointments
#
#  id                 :integer          not null, primary key
#  title              :string(255)      not null
#  doctor_id          :integer          not null
#  created_at         :datetime
#  updated_at         :datetime
#  email              :string(255)
#  fname              :string(255)
#  lname              :string(255)
#  startTime          :datetime
#  endTime            :datetime
#  office_hour        :boolean          default(FALSE)
#  appointment_status :string(255)      default("Pending")
#  phone_number       :string(255)
#  country_code       :string(255)
#

require 'mandrill'

class Appointment < ActiveRecord::Base
  validates :title, :startTime, :doctor, :appointment_status, presence: true
  validates :email, :email => true, allow_nil: true
  
  belongs_to :doctor
  
  def self.send_email_reminders
    appointments = Appointment.where({
      startTime: Date.tomorrow.midnight..Date.tomorrow.tomorrow.midnight,
      office_hour: false,
      appointment_status: "Confirmed"
    })
    appointments.each do |appointment|
      subject = "Recordatorio cita #{doctor.name}"
      body = "<p>Hola #{appointment.full_name},</p>"
      body += "<p>Recuerde que mañana tiene cita a las #{appointment.time} con #{appointment.doctor.name}.</p>"
      body += "<p>Enviado por Assisster</p>"
      appointment.send_email(subject, body)
    end
  end
  
  def full_name
    "#{fname} #{lname}"
  end
  
  def date
    self.startTime.strftime("%d/%m")
  end
  
  def time
    self.startTime.strftime("%H:%M")
  end
  
  def full_phone
    "#{country_code}#{phone_number}"
  end
    
  def send_email(subject, body)
    html_msg = body
    doctor = self.doctor
    begin
      mandrill = Mandrill::API.new ENV["MANDRILL_API_KEY"]
      message = {
        "html"=>html_msg,
        "text"=>body,
        "subject"=>subject,
        "from_email"=>doctor.email,
        "from_name"=>doctor.name,
        "to"=>
          [{"email"=>self.email,
              "name"=>self.full_name,
              "type"=>"to"}],
        "headers"=>{"Reply-To"=>doctor.email}
     }
     async = true
     result = mandrill.messages.send message, async
   rescue Mandrill::Error => e
     puts "A mandrill error occurred: #{e.class} - #{e.message}"
   end
  end
end
